Classes/DataProcessing/DataProcessingTrait.php

```diff
@@ -49,12 +64,18 @@ protected function isMenuProcessor(): bool
      * Removes recursively "data" in children nodes
      *
      * @param array $children
+     * @param bool $removeAll
+     * @param array $keepItems
      * @param string $nodeName
      */
-    private function removeDataInChildrenNodes(array &$children, string $nodeName = 'children'): void
+    private function removeDataInChildrenNodes(array &$children, bool $removeAll, array $keepItems, string $nodeName = 'children'): void
```

Comment:
we can pass only 1 new param (`$itemsToKeep`), if not empty array then filter out fields, if empty just unset whole data (default)